skimage/_shared/testing.py

```diff
@@ -132,28 +137,28 @@ def color_check(plugin, fmt='png'):
     All major input types should be handled as ubytes and read
     back correctly.
     """
-    img = img_as_ubyte(data.chelsea())
+    img = rescale_to_ubyte(data.chelsea())
     r1 = roundtrip(img, plugin, fmt)
     testing.assert_allclose(img, r1)
 
     img2 = img > 128
     r2 = roundtrip(img2, plugin, fmt)
     testing.assert_allclose(img2, r2.astype(bool))
 
-    img3 = img_as_float(img)
+    img3 = rescale_to_float(img, legacy_float_range=True)
```

Comment:
In cases like this, `legacy_float_range=True` wouldn't be necessary, `img` is boolean, but for internal stuff I chose to be on the safe side and just add `legacy_float_range=True`. That was a lot faster than me going through every case and trying to determine if the input could be an signed integer, and potentially missing an edge case and introducing an unnecessary error.